rotkehlchen/db/filtering.py

```diff
@@ -998,15 +993,64 @@ def make(
         return filter_query
 
 
+class EvmEventFilterQuery(HistoryEventFilterQuery):
+    @classmethod
+    def make(
+            cls,
+            and_op: bool = True,
+            order_by_rules: Optional[list[tuple[str, bool]]] = None,
+            limit: Optional[int] = None,
+            offset: Optional[int] = None,
+            from_ts: Optional[Timestamp] = None,
+            to_ts: Optional[Timestamp] = None,
+            assets: Optional[tuple[Asset, ...]] = None,
+            event_types: Optional[list[HistoryEventType]] = None,
+            event_subtypes: Optional[list[HistoryEventSubType]] = None,
+            exclude_subtypes: Optional[list[HistoryEventSubType]] = None,
+            location: Optional[Location] = None,
+            location_label: Optional[str] = None,
+            ignored_ids: Optional[list[str]] = None,
+            null_columns: Optional[list[str]] = None,
+            event_identifiers: Optional[list[bytes]] = None,
+            exclude_ignored_assets: bool = False,
+            counterparties: Optional[list[str]] = None,
+    ) -> 'EvmEventFilterQuery':
+        filter_query = super().make(
+            and_op=and_op,
+            order_by_rules=order_by_rules,
+            limit=limit,
+            offset=offset,
+            from_ts=from_ts,
+            to_ts=to_ts,
+            assets=assets,
+            event_types=event_types,
+            event_subtypes=event_subtypes,
+            exclude_subtypes=exclude_subtypes,
+            location=location,
+            location_label=location_label,
+            ignored_ids=ignored_ids,
+            null_columns=null_columns,
+            event_identifiers=event_identifiers,
+            exclude_ignored_assets=exclude_ignored_assets,
+        )
+        filter_query = cast(EvmEventFilterQuery, filter_query)
+        if counterparties is not None:
+            filter_query.filters.append(
+                DBPCounterpartyFilter(and_op=True, counterparties=counterparties),
+            )
+
+        return filter_query
+
+
 @dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
-class DBProtocolsFilter(DBFilter):
-    """Protocols should be not empty. Otherwise, it can break the query's logic"""
-    protocols: list[str]
+class DBPCounterpartyFilter(DBFilter):
```

Comment:
This is a terrible name. What is `P` doing in there? Write a full name, for whatever this filter is supposed to be